web/src/pages/api/public/scores.ts

```diff
@@ -1,167 +1,129 @@
-import { ScoreSource, prisma } from "@langfuse/shared/src/db";
-import { Prisma, type Score } from "@langfuse/shared/src/db";
-import { type NextApiRequest, type NextApiResponse } from "next";
-import { z } from "zod";
-import { cors, runMiddleware } from "@/src/features/public-api/server/cors";
-import { verifyAuthHeaderAndReturnScope } from "@/src/features/public-api/server/apiAuth";
-import { paginationZod } from "@langfuse/shared";
-import {
-  ScoreBody,
-  eventTypes,
-  ingestionBatchEvent,
-  stringDate,
-} from "@langfuse/shared";
+import { prisma } from "@langfuse/shared/src/db";
+import { Prisma } from "@langfuse/shared/src/db";
+import { eventTypes, ingestionBatchEvent } from "@langfuse/shared";
+import * as Sentry from "@sentry/node";
 import { v4 } from "uuid";
 import {
   handleBatch,
   handleBatchResultLegacy,
 } from "@/src/pages/api/public/ingestion";
-import { isPrismaException } from "@/src/utils/exceptions";
-
-const operators = ["<", ">", "<=", ">=", "!=", "="] as const;
-
-const ScoresGetSchema = z
-  .object({
-    ...paginationZod,
-    userId: z.string().nullish(),
-    name: z.string().nullish(),
-    fromTimestamp: stringDate,
-    source: z.nativeEnum(ScoreSource).nullish(),
-    value: z.coerce.number().nullish(),
-    operator: z.enum(operators).nullish(),
-    scoreIds: z
-      .string()
-      .transform((str) => str.split(",").map((id) => id.trim())) // Split the comma-separated string
-      .refine((arr) => arr.every((id) => typeof id === "string"), {
-        message: "Each score ID must be a string",
-      })
-      .nullish(),
-  })
-  .strict(); // Use strict to give 400s on typo'd query params
-
-export default async function handler(
-  req: NextApiRequest,
-  res: NextApiResponse,
-) {
-  await runMiddleware(req, res, cors);
-
-  // CHECK AUTH
-  const authCheck = await verifyAuthHeaderAndReturnScope(
-    req.headers.authorization,
-  );
-  if (!authCheck.validKey)
-    return res.status(401).json({
-      message: authCheck.error,
-    });
-  // END CHECK AUTH
-
-  if (req.method === "POST") {
-    try {
-      console.log(
-        "trying to create score, project ",
-        authCheck.scope.projectId,
-        ", body:",
-        JSON.stringify(req.body, null, 2),
-      );
+import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
+import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
+import {
+  GetScoresData,
+  GetScoresQuery,
+  GetScoresResponse,
+  PostScoresBody,
+  PostScoresResponse,
+  type ValidatedGetScoresData,
+} from "@/src/features/public-api/types/scores";
 
+export default withMiddlewares({
+  POST: createAuthedAPIRoute({
+    name: "Create Score",
+    bodySchema: PostScoresBody,
+    responseSchema: PostScoresResponse,
```

Comment:
as this routes via the batch endpoint, we do not get a scores object back which is not nice. shall we add additional logic here?

Reply:
ignore for now, could add later

Reply:
but cover in tests that it is void